PF/PF-Assignment2/FixedProductPromotion.java

```diff
@@ -0,0 +1,151 @@
+import java.util.HashMap;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.text.ParseException;
+
+public class FixedProductPromotion extends ShoppingCart implements Promotion {
+	
+	private HashMap<Long, Double> fixedDiscount;
```

Comment:
No need for this hashMap, just add a class level variable 'fixDiscount' will be applicable for each product. write getter and setter for this.